### UNIMODAL PROCESSING - VISION
oak:
  ros__parameters:
    camera:
      i_enable_imu: true
      i_enable_ir: true
      i_nn_type: spatial
      i_pipeline_type: RGBD
    stereo:
      i_subpixel: true
    nn:
      i_nn_config_path: depthai_ros_driver/yolo
      i_disable_resize: true
    rgb:
      i_preview_size: 416

clip_scene_rec:
  ros__parameters:
    num_est_interval_samples: 15
    scene_labels: ['campus','courtyard','lab','lobby']
    scene_descriptions: ['a picture of a public outdoor area with high social activity',
                         'a picture of a private outdoor area with low social activity',
                         'a picture of a private indoor area with low social activity',
                         'a picture of a public indoor area with high social activity']
    clip_model: 'ViT-L/14'

### UNIMODAL PROCESSING - AUDITION
audio_acq_node:
  ros__parameters:
    n_channels: 16
    src: 'plughw:1,0'
    sample_rate: 44100
    hop_size: 44100
    frame_size: 44100 # publish 1 second every second
    microphone_frame_id: 'philbart/base_link'

audio_scene_rec:
  ros__parameters:
    n_channels: 16
    sample_rate: 441000
    downsample_rate: 32000
    frame_size: 44100
    scene_size: 44100
    scene_index: [0]
    labels: ['airport', 'bus', 'metro', 'metro_station', 'park', 'public_square', 'shopping_mall', 'street_pedestrian', 'street_traffic', 'tram']
    scene_est_interval: 1.0
    wandb_id: 'fjycgepm'

    # Model path params
    project_name: DCASE23_Task1
    wandb_id: 'c0a7nzin'

    # Mel pre-processing parameters
    resample_rate: 32000
    window_size: 3072
    hop_size: 500
    n_fft: 4096
    n_mels: 256
    freqm: 48
    timem: 0
    fmin: 0
    fmax: 0
    fmin_aug_range: 1
    fmax_aug_range: 1000

    # Model parameters
    n_classes: 10
    in_channels: 1
    base_channels: 32
    channels_multiplier: 2.3
    expansion_rate: 3


### MULTIMODAL PROCESSING - SCENE RECOGNITION
/bayes_fused_scene_est:
  ros__parameters:

    scene_labels: ['campus','courtyard','lab','lobby']
    scene_prior: [.25,.25,.25,.25]

    sensor_names: ['clip','dcase23']

    clip:
      obs_labels: ['campus','courtyard','lab','lobby']
      sensor_model_coeffs: [35272., 3382., 1., 1.,
                            154., 42755., 1., 1.,
                            1., 54., 41930., 211.,
                            1., 15., 4906., 36521.]
      topic: 'clip_scene_category'

    dcase23:
      obs_labels: ['airport', 'bus', 'metro', 'metro_station','park','public_square','shopping_mall','street_pedestrian','street_traffic','tram']
      sensor_model_coeffs: [8., 17., 49., 22., 191., 7., 1., 1., 44., 1948.,
                            4., 305., 5., 8., 765., 1., 1., 2., 22., 1362.,
                            22., 156., 2125., 33., 79., 2., 1., 2., 66., 21.,
                            457., 57., 205., 481., 977., 2., 70., 17., 20., 166.]
      topic: 'audio_scene_category'